tests/input/examples/schema_definition-native-array-rgb.yaml

```diff
@@ -0,0 +1,44 @@
+id: https://example.org/arrays
+name: arrays-temperature-example
+title: Array Temperature Example
+description: |-
+  Example LinkML schema to demonstrate a 3D DataArray of temperature values with labeled axes
+license: MIT
+
+prefixes:
+  linkml: https://w3id.org/linkml/
+  wgs84: http://www.w3.org/2003/01/geo/wgs84_pos#
+  example: https://example.org/
+
+default_prefix: example
+
+imports:
+  - linkml:types
+
+classes:
+
+  RGBImage:
+    attributes:
+      rgb:
+        range: float
+        array_info:
+          # NPtyping: NDArray[Shape["* x, * y, 3 rgb"]
+          exact_dimensions: 3
+          dimensions_info:
+            x:
+            y:
+            rgb:
+              exact_cardinality: 3
+              description: r, g, b values
+              annotations:
+                names: "[red, green, blue]"
+  SquareDataset:
```

Comment:
specifically on this idea of expressing a square dataset, where we want to specify "array can be any cardinality where the cardinality of `x` and `y` are equal, would it be possible to specify cardinality and dimension as an expression like `exact_dimensions: "prefix:Dimension * 2"` or something like that? or maybe that's a v2 kind of thing

Reply:
discussed - we do not want to make a DSL right now